pkg/kube/wait.go

```diff
@@ -188,12 +188,24 @@ func (w *waiter) serviceReady(s *corev1.Service) bool {
 	}
 
 	// Make sure the service is not explicitly set to "None" before checking the IP
-	if (s.Spec.ClusterIP != corev1.ClusterIPNone && s.Spec.ClusterIP == "") ||
-		// This checks if the service has a LoadBalancer and that balancer has an Ingress defined
-		(s.Spec.Type == corev1.ServiceTypeLoadBalancer && s.Status.LoadBalancer.Ingress == nil) {
-		w.log("Service does not have IP address: %s/%s", s.GetNamespace(), s.GetName())
+	if s.Spec.ClusterIP != corev1.ClusterIPNone && s.Spec.ClusterIP == "" {
```

Comment:
Can you add back some logging here in this `if` block to explain why it isn't ready?

Reply:
Sorry, my bad. I'm adding back the original message. 
However I think it can be made a little more specific, for example: `w.log("Service does not have cluster IP address: %s/%s", s.GetNamespace(), s.GetName())`

Reply:
Feel free to make it more specific now that the logic is separated out!

Reply:
I've Just pushed a commit with a little improvement of the logs.
If for your is fine, for me it's done.